### ISSUE TITLE
Bug? 2nd compound index containing date doesn't sort

### ISSUE THREAD

**Issue by webmasterkai** [rethinkdb/rethinkdb#5403]
I have a table with two compound indexes. Each index ends with a date field. The first index will sort correctly. The 2nd index does not.

Steps to reproduce:

```javascript
// Create new table.
r.db('foo').tableCreate('test')
// Insert 2000 new records each with a date about one min apart.
r.db('foo').table('test')
  .insert(r.range(2000).map({
    date: r.now().sub(r.row.mul(60)),
    foo: 'mar',
    me: 'too',
  }))
// Create first compound index with a date field.
r.db('foo').table('test')
  .indexCreate('normal', [ r.row('foo'), r.row('date') ])
// Sort the list based on our compound index.
r.db('foo').table('test')
  .orderBy({ index: r.desc('normal') })
  .map(r.now().toEpochTime().sub(r.row('date').toEpochTime()).round())
// Everything looks normal...

// Create a 2nd index with a date field.
r.db('foo').table('test')
  .indexCreate('strange', [ r.row('me'), r.now('date') ])
// Run query like above with a sort.
r.db('foo').table('test')
  .orderBy({ index: r.desc('strange') })
  .map(r.now().toEpochTime().sub(r.row('date').toEpochTime()).round())
// WTF?
```

Maybe I'm not thinking of something obvious?
```
Running rethinkdb 2.2.2 (CLANG 4.2 (clang-425.0.28))...
Running on Darwin 15.3.0 x86_64
```
Thanks!

[edit] The first index can be *any* kind of index.

**Comment by Tryneus** [comment]
There appears to be a typo in that code that could explain the problem, `r.now('date')` should be `r.row('date')`.  Super easy to overlook.

**Comment by webmasterkai** [comment]
OMG. I love you @Tryneus! Can't thank you enough! Thank you, thank you, thank you. It works as expected if I spell things correctly. Imagine that!

**Comment by webmasterkai** [comment]
I've noticed other functions throw an error if they get too many arguments. Should `r.now()` throw if it gets an argument?

**Comment by Tryneus** [comment]
It should, but doesn't, but that is probably due to how it's handled on the server - it isn't put into the term tree with the rest of the query, but it is instead rewritten to a literal value before we run the query.

It's entirely possible that `r.now` used to check the arity and I screwed it up when I rewrote that code a few months back.  Either way, it should be a simple fix.

Thanks for reporting this, @webmasterkai, and glad to help!
